dmsrc/sound_len.dm

```diff
@@ -0,0 +1,34 @@
+// Provided a static RSC file path or a raw text file path, returns the duration of the file in deciseconds as a float.
+/proc/rustg_sound_length(file_path)
+	var/static/list/sound_cache
+	if(isnull(sound_cache))
+		sound_cache = list()
+
+	. = 0
+
+	if(!istext(file_path))
+		if(!isfile(file_path))
+			CRASH("rustg_sound_length error: Passed non-text object")
+
+		if(length("[file_path]"))
+			file_path = "[file_path]"
```

Comment:
what is this code doing

Reply:
This checks if an RSC path is a real file. If it cant be stringified it isn't a file on the disk.

Reply:
then comment that